tests/testthat/test-summarisePopulationCharacteristics.R

```diff
@@ -0,0 +1,117 @@
+  # Load mock database ----
+  cdm <- cdmEunomia()
+
+  # Check that works ----
+  expect_no_error(summarisedPopulation <- summarisePopulationCharacteristics(
+    cdm = cdm)
+    )
+  expect_true(inherits(summarisedPopulation,"summarised_result"))
+  expect_true(all(summarisedPopulation |>
+                    dplyr::select("strata_name") |>
+                    dplyr::distinct() |>
+                    dplyr::pull() ==
+                    c("overall")))
+  expect_true(all(summarisedPopulation |>
+                    dplyr::filter(variable_name == "Number records") |>
+                    dplyr::select("estimate_value") |>
+                    dplyr::pull() ==
+                    2694))
+  expect_true(all(summarisedPopulation |>
+                    dplyr::filter(variable_name == "Cohort start date" & estimate_name == "min") |>
+                    dplyr::select("estimate_value") |>
+                    dplyr::pull() ==
+                    "1908-09-22"))
+  expect_true(summarisedPopulation |>
+                dplyr::filter(variable_name == "Age at end") |>
+                dplyr::tally() |>
+                dplyr::pull() !=
+                0)
+
+  # Add date range
+  expect_no_error(summarisedPopulation <- summarisePopulationCharacteristics(
+    cdm = cdm,
+    studyPeriod = c("1900-01-01", "2010-01-01"))
+  )
+  expect_true(inherits(summarisedPopulation,"summarised_result"))
+  expect_true(all(summarisedPopulation |>
+                    dplyr::select("strata_name") |>
+                    dplyr::distinct() |>
+                    dplyr::pull() ==
+                    c("overall")))
+  expect_true(all(summarisedPopulation |>
+                    dplyr::filter(variable_name == "Number records") |>
+                    dplyr::select("estimate_value") |>
+                    dplyr::pull() ==
+                    2694))
+  expect_true(all(summarisedPopulation |>
+                    dplyr::filter(variable_name == "Cohort end date" & estimate_name == "max") |>
+                    dplyr::select("estimate_value") |>
+                    dplyr::pull() ==
+                    "2010-01-01"))
+
+  # Add sex and age group strata
+  expect_no_error(summarisedPopulation <- summarisePopulationCharacteristics(
+    cdm = cdm,
+    studyPeriod = c("1950-01-01", NA),
+    sex = TRUE,
+    ageGroup = list(c(0,20),c(21,150)))
+    )
+  expect_true(inherits(summarisedPopulation,"summarised_result"))
+  expect_true(all(summarisedPopulation |>
+                    dplyr::select("strata_name") |>
+                    dplyr::distinct() |>
+                    dplyr::pull() ==
+                    c("overall", "sex", "age_group", "sex &&& age_group")))
+  expect_true(all(summarisedPopulation |>
+                    dplyr::filter(variable_name == "Number records") |>
+                    dplyr::select("estimate_value") |>
+                    dplyr::pull() ==
+                    c(2693,1372,1321,2521,172,1271,101,1250,71)))
```

Comment:
This is particular of eunomia, and you believe that the result that it gives you is right. Could we test this with a real example where you really know the truth?